OneCommunityGlobal/HGNRest
# Ivy-apply-MissHrCoreteam-blue-square-email

# Description
<img width="675" alt="Screenshot 2024-05-23 at 10 43 49 AM" src="https://github.com/OneCommunityGlobal/HGNRest/assets/102696148/4d2932aa-ad7c-45e0-8ab6-6c596a36344a">

## Main changes explained:
- Modify userHelper.js
…

## How to test:
1. check into current branch
2. Gmail API
Before testing, you need to set up your own test email id to send the mail notification.
This is only for test purposes. Once tested you can revert these changes..

To send an email using the Gmail API with OAuth2 and nodemailer, follow the video below and set it up.
https://www.youtube.com/watch?v=-rcRf7yswfM&t=10s&ab_channel=MafiaCodes

Fill in the these variable values from the above setup in the .env file of the backend.
REACT_APP_EMAIL =
REACT_APP_EMAIL_CLIENT_ID =
REACT_APP_EMAIL_CLIENT_SECRET =
REACT_APP_EMAIL_REFRESH_TOKEN =
REACT_APP_EMAIL_CLIENT_REDIRECT_URI = https://developers.google.com/oauthplayground
sendEmail = true
3. comment this to run every minute
<img width="697" alt="Screenshot 2024-03-31 at 1 49 23 PM" src="https://github.com/OneCommunityGlobal/HGNRest/assets/102696148/f16a7de4-f12a-41ab-8536-1e3f69fb916e">

- add test user email(role is Core Team) here:

<img width="668" alt="Screenshot 2024-03-31 at 1 29 58 PM" src="https://github.com/OneCommunityGlobal/HGNRest/assets/102696148/ba80a03e-357e-4932-a92f-3715ffd1bfaa">

- change the email addressyou wanna receive the blue square email
<img width="915" alt="Screenshot 2024-05-23 at 10 46 28 AM" src="https://github.com/OneCommunityGlobal/HGNRest/assets/102696148/3ffde3d4-3dfb-4cb2-be5a-9a638046a425">


4. the email be like this
<img width="1227" alt="Screenshot 2024-05-23 at 10 40 48 AM" src="https://github.com/OneCommunityGlobal/HGNRest/assets/102696148/dfcd90b8-af1b-4e08-bdb6-6da6f05d0dd1">


## Note:
Include the information the reviewers need to know.
